archive/android_img.ksy

```diff
@@ -0,0 +1,116 @@
+meta:
+  id: android_img
+  title: Android Boot Image
+  license: CC0-1.0
+  file-extension: img
+  endian: le
+doc-ref: https://source.android.com/devices/bootloader/boot-image-header
+seq:
+  - id: magic
+    contents: ANDROID!
+  - id: kernel
+    type: load
+  - id: ramdisk
+    type: load
+  - id: second
+    type: load
+  - id: tags_load
+    type: u4
+  - id: page_size
+    type: u4
+  - id: header_version
+    type: u4
+  - id: os_version
+    type: os_version
+  - id: name
+    type: strz
+    size: 16
+    encoding: ASCII
+  - id: cmdline
+    type: strz
+    size: 512
+    encoding: ASCII
+  - id: sha
+    size: 32
+  - id: extra_cmdline
+    type: strz
+    size: 1024
+    encoding: ASCII
+  - id: recovery_dtbo
+    type: load_offset
+    if: header_version > 0
+  - id: boot_header_size
+    type: u4
+    if: header_version > 0
+  - id: dtb
+    type: load_long
+    if: header_version > 1
+instances:
+  base:
+    value: kernel.addr - 0x00008000
+  kernel_offset:
+    value: kernel.addr - base
+  ramdisk_offset:
+    value: ramdisk.addr - base
+  second_offset:
+    value: second.addr - base
+  tags_offset:
+    value: tags_load - base
+  recovery_dtbo_offset:
+    value: recovery_dtbo.offset
```

Comment:
I tried your spec on a public sample file [`boot.img`](https://github.com/cfig/Android_boot_image_editor/blob/master/src/test/resources/boot.img) and the parsing failed due to nullability of `recovery_dtbo` field (you are missing the version check ensuring that the `recovery_dtbo` is not `null`):
```suggestion
  recovery_dtbo_offset:
    value: recovery_dtbo.offset
    if: header_version > 0
```

Reply:
I've removed recovery_dtbo_offset from instances. Because this is a different kind of offset. recovery_dtbo.offset is offset for recovery_dtbo file in boot.img. Other offsets are memory loading offsets relative to base addr.